File: PdfClown-NetStandard/CliSamples/org/pdfclown/samples/cli/AcroFormFillingSample.cs
using org.pdfclown.documents;
using org.pdfclown.documents.contents.composition;
using org.pdfclown.documents.contents.entities;
using org.pdfclown.documents.contents.fonts;
using org.pdfclown.documents.contents.xObjects;
using org.pdfclown.documents.interaction.annotations;
using org.pdfclown.documents.interaction.forms;
using org.pdfclown.files;

using System;
using System.Collections.Generic;
using System.Drawing;

namespace org.pdfclown.samples.cli
{
  /**
    <summary>This sample demonstrates how to fill AcroForm fields of a PDF document.</summary>
  */
  public class AcroFormFillingSample
    : Sample
  {
    public override void Run(
      )
    {
      // 1. Opening the PDF file...
      string filePath = PromptFileChoice("Please select a PDF file");
      File file = new File(filePath);
      Document document = file.Document;

      // 2. Get the acroform!
      Form form = document.Form;
      if(!form.Exists())
      {Console.WriteLine("\nNo acroform available.");}
      else
      {
        // 3. Filling the acroform fields...
        int mode;
        try
        {
          IDictionary<string,string> options = new Dictionary<string,string>();
          options["0"] = "Automatic filling";
          options["1"] = "Manual filling";
          mode = Int32.Parse(PromptChoice(options));
        }
        catch
        {mode = 0;}
        switch(mode)
        {
          case 0: // Automatic filling.
            Console.WriteLine("\nAcroform is being filled with random values...\n");

            foreach(Field field in form.Fields.Values)
            {
              String value;
              if(field is RadioButton)
              {value = field.Widgets[0].Value;} // Selects the first widget in the group.
              else if(field is ChoiceField)
              {value = ((ChoiceField)field).Items[0].Value;} // Selects the first item in the list.
              else
              {value = field.Name;} // Arbitrary value (just to get something to fill with).
              field.Value = value;
            }
            break;
          case 1: // Manual filling.
            Console.WriteLine("\nPlease insert a value for each field listed below (or type 'quit' to end this sample).\n");

            foreach(Field field in form.Fields.Values)
            {
              Console.WriteLine("* " + field.GetType().Name + " '" + field.FullName + "' (" + field.BaseObject + "): ");
              Console.WriteLine("    Current Value:" + field.Value);
              string newValue = PromptChoice("    New Value:");
              if(newValue != null && newValue.Equals("quit"))
                break;

              field.Value = newValue;
            }
            break;
        }
      }

      // 4. Serialize the PDF file!
      Serialize(file);
    }
  }
}

File: java/pdfclown.samples.cli/src/org/pdfclown/samples/cli/AcroFormFillingSample.java
package org.pdfclown.samples.cli;

import java.util.HashMap;
import java.util.Map;

import org.pdfclown.documents.Document;
import org.pdfclown.documents.interaction.forms.ChoiceField;
import org.pdfclown.documents.interaction.forms.Field;
import org.pdfclown.documents.interaction.forms.Form;
import org.pdfclown.documents.interaction.forms.RadioButton;
import org.pdfclown.files.File;

/**
  This sample demonstrates <b>how to fill AcroForm fields</b> of a PDF document.

  @author Stefano Chizzolini (http://www.stefanochizzolini.it)
  @since 0.1.1
  @version 0.1.2.1, 03/21/15
*/
public class AcroFormFillingSample
  extends Sample
{
  @Override
  public void run(
    )
  {
    // 1. Opening the PDF file...
    File file;
    {
      String filePath = promptFileChoice("Please select a PDF file");
      try
      {file = new File(filePath);}
      catch(Exception e)
      {throw new RuntimeException(filePath + " file access error.",e);}
    }
    Document document = file.getDocument();

    // 2. Get the acroform!
    Form form = document.getForm();
    if(!form.exists())
    {System.out.println("\nNo acroform available.");}
    else
    {
      // 3. Filling the acroform fields...
      int mode;
      try
      {
        Map<String,String> options = new HashMap<String,String>();
        options.put("0", "Automatic filling");
        options.put("1", "Manual filling");
        mode = Integer.parseInt(promptChoice(options));
      }
      catch(Exception e)
      {mode = 0;}
      switch(mode)
      {
        case 0: // Automatic filling.
          System.out.println("\nAcroform is being filled with random values...\n");

          for(Field field : form.getFields().values())
          {
            String value;
            if(field instanceof RadioButton)
            {value = field.getWidgets().get(0).getValue();} // Selects the first widget in the group.
            else if(field instanceof ChoiceField)
            {value = ((ChoiceField)field).getItems().get(0).getValue();} // Selects the first item in the list.
            else
            {value = field.getName();} // Arbitrary value (just to get something to fill with).
            field.setValue(value);
          }
          break;
        case 1: // Manual filling.
          System.out.println("\nPlease insert a value for each field listed below (or type 'quit' to end this sample).\n");

          for(Field field : form.getFields().values())
          {
            System.out.println("* " + field.getClass().getSimpleName() + " '" + field.getFullName() + "' (" + field.getBaseObject() + "): ");
            System.out.println("    Current Value:" + field.getValue());
            String newValue = promptChoice("    New Value:");
            if(newValue != null && newValue.equals("quit"))
              break;

            field.setValue(newValue);
          }
          break;
      }
    }

    // 4. Serialize the PDF file!
    serialize(file);
  }
}
